File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dic={}
    for p in participant:
        dic[p]=0
        
    for p in participant:
        dic[p]+=1
        
    for c in completion:
        dic[c]-=1
        
    for p in participant:
        if (dic[p]!=0):
            return p

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> map = new HashMap<>();
        
        for (String p : participant) {
            map.put(p, 0);
        }
        
        for (String p : participant) {
            map.put(p, map.get(p) + 1);
        }
        
        for (String c : completion) {
            map.put(c, map.get(c) - 1);
        }
        
        for (String p : participant) {
            if (map.get(p) != 0) {
                answer = p;
            }
        }
        
        return answer;
    }
}